File: android/src/main/java/com/azifacemobile/theme/OrientationScreen.java
package com.azifacemobile.theme;

import android.graphics.Typeface;

import com.azifacemobile.theme.abstracts.ViewStyle;
import com.azifacemobile.utils.Theme;
import com.facebook.react.bridge.ReactApplicationContext;

import org.json.JSONObject;

import javax.annotation.Nullable;

public class OrientationScreen extends ViewStyle {
  private final static String KEY = "orientationScreen";
  private final JSONObject target;
  private final Font font;
  private final Image image;

  public OrientationScreen(ReactApplicationContext context) {
    super(KEY);

    this.target = new Theme().getTarget(KEY);
    this.font = new Font(context);
    this.image = new Image(context, this.target);
  }

  @Override
  public int getForegroundColor() {
    return super.getForegroundColor("#026ff4");
  }

  @Nullable
  public Typeface getFont() {
    return this.font.getTypography(this.target, "font");
  }

  public int getIconImage() {
    return this.image.getSource("iconImage");
  }
}

File: ios/Theme/OrientationScreen.swift
public class OrientationScreen: ViewStyle {
  private static let KEY: String = "orientationScreen"
  private let target: NSDictionary?
  private let font: Font
  private let image: Image

  init() {
    self.target = Style().getTarget(OrientationScreen.KEY)
    self.font = Font()
    self.image = Image(target: self.target)

    super.init(key: OrientationScreen.KEY)
  }

  override func getForegroundColor() -> UIColor {
    return super.getForegroundColor(defaultColor: "#026ff4")
  }

  public func getFont() -> UIFont {
    return self.font.getTypography(theme: self.target, key: "font")
  }

  public func getIconImage() -> UIImage? {
    return self.image.getSource("iconImage")
  }
}
